src/components/blog/AddBlog.tsx

```diff
@@ -62,30 +89,85 @@ const AddBlog = () => {
     if (!heroPhoto) {
       return enqueueSnackbar("No here image selected.", { variant: "error" });
     }
-    let imageId;
-    let blogSlug;
-
-    blogStorage(heroPhoto, data.title).then((res) => {
-      ({ imageId, blogSlug } = res);
-      postBlog(blogSlug, {
-        ...data,
-        author: user.name,
-        imageId: imageId,
-        slug: blogSlug,
-        content: content,
-      })
-        .then(() => enqueueSnackbar("Blog card has been added"))
-        .catch((e) =>
-          enqueueSnackbar(`Error adding the blog card: ${e.message}`, {
-            variant: "error",
+
+    const authorName = user.name ? user.name : "Error";
```

Comment:
For conciseness do `authorName = user.name || "Error"`

However, instead of storing "Error" in the database I would just return `null` instead